keen/explorer
# Fix visualization selector

## What does this PR do? How does it affect users?

This fixes a few bugs:

* User is not able to run a query of a different data type without visualizations breaking.
* User is not able to switch between different chart types.

## How should this be tested?

* Run a count query in explorer. Add an interval, re-run the query. Observe that a chart gets rendered.
* Run a query in explorer that has multiple chart types (example: interval of a count). Switch between the different types of charts many times. Observe that the chart type changes to the appropriate visualization.

Step through the code line by line. Things to keep in mind as you review:
 - Are there any edge cases not covered by this code?
 - Does this code follow conventions (naming, formatting, modularization, etc) where applicable?

Fetch the branch and/or deploy to staging to test the following:
